python-stdlib/logging/logging.py

```diff
@@ -43,25 +44,40 @@ def __init__(self, level=NOTSET):
         self.level = level
         self.formatter = None
 
+    @staticmethod
+    def _default_formatter():
+        global _default_formatter
+        if _default_formatter is None:
+            _default_formatter = Formatter()
+        return _default_formatter
+
     def close(self):
         pass
 
+    def flush(self):
+        pass
+
     def setLevel(self, level):
         self.level = level
 
     def setFormatter(self, formatter):
         self.formatter = formatter
 
     def format(self, record):
-        return self.formatter.format(record)
+        if self.formatter:
+            fmt = self.formatter
+        else:
+            fmt = self._default_formatter()
+        return fmt.format(record)
 
 
 class StreamHandler(Handler):
     def __init__(self, stream=None):
+        super().__init__()
         self.stream = _stream if stream is None else stream
         self.terminator = "\n"
 
-    def close(self):
+    def flush(self):
```

Comment:
Should there still be a `close()` method which just calls `self.flush()`?

Reply:
Good point. Cpython does not have a `StreamHandler.close()` method, and so the inherited `close()` method doesn't do any output.